packages/types/src/api/web/searchFilter.ts

```diff
@@ -1,23 +1,57 @@
 import { FieldType } from "../../documents"
 import {
   EmptyFilterOption,
-  FilterGroupLogicalOperator,
-  SearchFilters,
+  UILogicalOperator,
+  BasicOperator,
+  RangeOperator,
+  ArrayOperator,
 } from "../../sdk"
 
-export type LegacyFilter = {
-  operator: keyof SearchFilters | "rangeLow" | "rangeHigh"
-  onEmptyFilter?: EmptyFilterOption
+type AllOr = {
+  operator: "allOr"
+}
+
+type OnEmptyFilter = {
+  onEmptyFilter: EmptyFilterOption
+}
+
+// TODO(samwho): this could be broken down further
+export type SearchFilter = {
+  operator:
+    | BasicOperator
+    | RangeOperator
+    | ArrayOperator
+    | "rangeLow"
+    | "rangeHigh"
+  // Field name will often have a numerical prefix when coming from the frontend,
+  // use the ColumnSplitter class to remove it.
   field: string
-  type?: FieldType
   value: any
+  type?: FieldType
   externalType?: string
+  noValue?: boolean
```

Comment:
What is this? I cannot see any usage of it

Reply:
The frontend sends it so I thought it would be worth representing in the type. Happy to remove it if you think it just adds confusion, though.

Reply:
Corollary: https://github.com/Budibase/budibase/pull/14818#discussion_r1810295475